suse/module-setup.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+check() {
+	[ -n "$hostonly" -a -c /sys/class/infiniband_verbs/uverbs0 ] && return 0
+	[ -n "$hostonly" ] && return 255
+	return 0
+}
+
+depends() {
+	return 0
+}
+
+install() {
+	inst /etc/rdma/mlx4.conf
+	inst /etc/rdma/modules/infiniband.conf
+	inst /etc/rdma/modules/iwarp.conf
+	inst /etc/rdma/modules/opa.conf
+	inst /etc/rdma/modules/rdma.conf
+	inst /etc/rdma/modules/roce.conf
+	inst /usr/lib/mlx4-setup.sh
+	inst_multiple lspci setpci awk sleep
+	inst_rules 60-rdma-persistent-naming.rules 70-persistent-ipoib.rules 75-rdma-description.rules 90-rdma-hw-modules.rules 90-rdma-ulp-modules.rules
+	inst_multiple -o \
+                  $systemdsystemunitdir/rdma-hw.target \
```

Comment:
Does suse use systemd in the initrd that it needs this stuff?

Reply:
We do have systemd in initrd.
We initially used the redhat rdma init script which was not working (at least because some files were not packaed the same way or even not installed).

As we have systemd available, I figured it would be cleaner to use the udev/systemd scripts rather than copy/patch the redhat ones.

Someone at SUSE tested this patch (before the persistent name rules) and it worked well

Reply:
Note: It was tested on top of v16.5.
I simply checked that dracut managed to generate an image on master

Reply:
Can you copy "*.rules" with the asterisk? It will ensure that any change in kernel-boot folder won't break SuSE dracut.

Reply:
Not sure about that.
Plus this is run on a full system so if it worked, it'd probably pull all the .service files, not on;y the rdma ones

Reply:
Hum, interesting. So I'd like to get to a world where we don't need explicit module loading..

Do you have a list of tasks that require explicit module loading that suse does from the initrd?

About the only one I know of today is network booting over ipoib..

Reply:
The use case that showed the bug here is kdump over IPoIB which should require roughly the same things